anitya/tests/lib/backends/test_cran.py

```diff
@@ -0,0 +1,115 @@
+# code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission
+# of Red Hat, Inc.
+#
+
+'''
+anitya tests for the CRAN backend.
+'''
+
+import unittest
+
+import anitya.lib.backends.cran as backend
+from anitya.db import models
+from anitya.lib.exceptions import AnityaPluginException
+from anitya.tests.base import DatabaseTestCase, create_distro
+
+
+BACKEND = 'CRAN (R)'
+
+
+class CranBackendtests(DatabaseTestCase):
+    """ custom backend tests. """
+
+    def setUp(self):
+        """ Set up the environnment, ran before every tests. """
+        super(CranBackendtests, self).setUp()
+
+        create_distro(self.session)
+        self.create_project()
+
+    def create_project(self):
+        """ Create some basic projects to work with. """
+        project = models.Project(
+            name='whisker',
+            homepage='http://github.com/edwindj/whisker',
+            backend=BACKEND,
+        )
+        self.session.add(project)
+        self.session.commit()
+
+        project = models.Project(
+            name='non-existent-name-that-cannot-exist',
+            homepage='https://cran.r-project.org/web/packages/non-existent-name-that-cannot-exist/',
+            backend=BACKEND,
+        )
+        self.session.add(project)
+        self.session.commit()
+
+    def test_cran_get_version(self):
+        """ Test the get_version function of the CRAN backend. """
+        pid = 1
+        project = models.Project.get(self.session, pid)
+        exp = '0.3.2'
+        obs = backend.CranBackend.get_version(project)
+        self.assertEqual(obs, exp)
+
+        pid = 2
+        project = models.Project.get(self.session, pid)
+        self.assertRaises(
+            AnityaPluginException,
+            backend.CranBackend.get_version,
+            project
+        )
+
+    def test_cran_get_versions(self):
+        """ Test the get_versions function of the CRAN backend. """
+        pid = 1
+        project = models.Project.get(self.session, pid)
+        exp = ['0.1', '0.3.2']
+        obs = backend.CranBackend.get_ordered_versions(project)
+        self.assertEqual(obs, exp)
+
+        pid = 2
+        project = models.Project.get(self.session, pid)
+        self.assertRaises(
+            AnityaPluginException,
+            backend.CranBackend.get_version,
+            project
+        )
+
+    def test_cran_check_feed(self):
+        """ Test the check_feed method of the CRAN backend. """
+        generator = backend.CranBackend.check_feed()
+        items = list(generator)
+
+        self.assertEqual(items[0], (
+            'xtractomatic',
+            'https://github.com/rmendels/xtractomatic',
+            'CRAN (R)', '3.4.2'))
+        self.assertEqual(items[1], (
+            'FedData',
+            'https://github.com/ropensci/FedData',
+            'CRAN (R)', '2.5.2'))
+        # etc...
+
+
+if __name__ == '__main__':
```

Comment:
Could you remove this? I've been meaning to remove them from all the existing tests since we use pytest as the test runner.